File: account-server-golang/server_api/UNREGISTER_GAMESERVER_INSTANCE.go
package server_api

import (
	"account-server/gameservers"
	"account-server/utils"
	"encoding/json"
	"errors"
)

type request_UNREGISTER_GAMESERVER_INSTANCE struct {
	Request_id string
	Lobby_id   string
}

// UNREGISTER_GAMESERVER_INSTANCE:
// 1. El controller (del gameserver node) envia el lobby_id.
// 2. El servidor elimina la entrada con lobby_id de la lista de gameserver instances.
func UNREGISTER_GAMESERVER_INSTANCE(message []byte, sendResponse func([]byte)) error {

	var r request_UNREGISTER_GAMESERVER_INSTANCE

	// Comprueba que estan todos los parametros requeridos.
	{
		var m map[string]any
		err := json.Unmarshal(message, &m)
		if err != nil {
			return err
		}
		if !utils.ContainsAllKeys(m, []string{"request_id", "lobby_id"}) {
			err = errors.New("UNREGISTER_GAMESERVER_INSTANCE bad input parameters")
			return err
		}
	}

	err := json.Unmarshal(message, &r)
	if err != nil {
		return err
	}

	gameservers.RemoveGameServer(r.Lobby_id)

	response, err := json.Marshal(map[string]any{
		"request_id": r.Request_id,
		"lobby_id":   r.Lobby_id,
	})

	if err != nil {
		return err
	}

	sendResponse(response)

	return err
}

File: account-server/server_api/UNREGISTER_GAMESERVER_INSTANCE.py

# UNREGISTER_GAMESERVER_INSTANCE:
# 1. El controller (del gameserver node) envia el lobby_id.
# 2. El servidor elimina la entrada con lobby_id de la lista de gameserver instances.
async def UNREGISTER_GAMESERVER_INSTANCE(req, ws, send_message, gameservers_info):

    # Aqui se debe comprobar que la req es correcta y en caso contrario se ignora sin enviar nada al cliente.
    if not all(k in req for k in ('request_id', 'lobby_id')):
        print("ERROR: UNREGISTER_GAMESERVER_INSTANCE bad input parameters.")
        return

    lobby_id = req['lobby_id']
    gsinstances = gameservers_info['instances']
    gsinstances.pop(lobby_id, None)

    response = {
        'request_id': req['request_id'],
        'lobby_id': lobby_id
    }

    await send_message(response, ws)   # send response